src/components/case_manager/PersonTable.tsx

```diff
@@ -0,0 +1,398 @@
+        }),
+      }}
+    >
+      <div
+        style={{
+          display: "flex",
+          justifyContent: "space-between",
+          width: "100%",
+        }}
+      >
+        <h2 className={styles.tableTitle}>{type}</h2>
+        <button onClick={onOpenInput} className={styles.addButton}>
+          Add {type}
+        </button>
+      </div>
+      <div
+        style={{
+          display: "flex",
+          justifyContent: "space-between",
+          width: "100%",
+        }}
+      >
+        <Typography color="inherit" variant="subtitle1" component="div">
+          {numSelected > 0
+            ? `Selected: ${numSelected}/${totalRows}`
+            : `Total: ${totalRows}`}
+        </Typography>
+
+        {numSelected > 0 ? (
+          <Tooltip title="Delete">
+            <IconButton>
+              <DeleteIcon />
+            </IconButton>
+          </Tooltip>
+        ) : (
+          <TextField
+            id="search-input"
+            placeholder="Search"
+            InputProps={{
+              startAdornment: (
+                <InputAdornment position="start">
+                  <SearchIcon />
+                </InputAdornment>
+              ),
+            }}
+            variant="standard"
+          />
+        )}
+      </div>
+    </Toolbar>
+  );
+}
+
+interface EnhancedTableInputProps<Column> {
+  inputCells: Column[];
+  type: "Student" | "Staff";
+  onCloseInput: () => void;
+}
+
+function EnhancedTableInput<Column extends HeadCell>({
+  inputCells,
+  type,
+  onCloseInput,
+}: EnhancedTableInputProps<Column>) {
+  return (
+    <TableRow>
+      <TableCell padding="checkbox" align="center">
+        <button onClick={onCloseInput} className={styles.closeButton}>
+          <CloseIcon />
+        </button>
+      </TableCell>
+      {inputCells.map((inputCell, idx) => {
+        return inputCell.hasInput ? (
+          <TableCell key={inputCell.id} align={"left"}>
+            <TextField
+              label={inputCell.label}
+              autoFocus={idx === 0}
+              required
+              size="small"
+              inputProps={{
+                form: "table_input_form",
+                name: inputCell.id,
+              }}
+            />
+          </TableCell>
+        ) : null;
+      })}
+      <TableCell>
+        <button
+          type="submit"
+          form="table_input_form"
+          className={styles.addButton}
+        >
+          Add {type}
+        </button>
+      </TableCell>
+    </TableRow>
+  );
+}
+
+const StyledTableRow = styled(TableRow)(() => ({
+  "&:nth-of-type(odd)": {
+    backgroundColor: "#F8F9FA",
+  },
+  "&.MuiTableRow-hover:hover": {
+    backgroundColor: "lightgray",
+  },
+  "&.Mui-selected": {
+    backgroundColor: "lightblue",
+  },
+}));
+
+interface EnhancedTableProps<Person, Column> {
+  people: Person[];
+  onSubmit: (event: React.FormEvent<HTMLFormElement>) => void;
+  headCells: Column[];
+  type: "Student" | "Staff";
```

Comment:
can this be inferred based on `people`?

Reply:
We could use the `isStudent` typeguard function on the first element of the `people` array to define `type`. But the issue is if there are no elements in `people` yet, how would we know to render the students or staff version?